integration/spark/src/main/scala/org/apache/spark/sql/carbondata/execution/datasources/SparkCarbonFileFormat.scala

```diff
@@ -86,6 +87,11 @@ class SparkCarbonFileFormat extends FileFormat
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
     val conf = sparkSession.sessionState.newHadoopConf()
+    if (options.isEmpty && files.isEmpty) {
```

Comment:
remove this code

Reply:
path is removed from options from 3.1, so this change is required to throw the proper exception

Reply:
done